contrib_bots/lib/gcalendar.py

```diff
@@ -0,0 +1,329 @@
+           '@gcalendar <event_title> | <date>\n'
+           '```\n'
+           'Please notice that pipes (`|`) *cannot* be used in the input parameters.\n\n'
+           'Here are some usage examples:\n'
+           '```\n'
+           '@gcalendar Meeting with John | 2017/03/14 13:37 | 2017/03/14 15:00:01 | EDT\n'
+           '@gcalendar Comida | en 10 minutos | en 2 horas\n'
+           '@gcalendar Trip to LA | tomorrow\n'
+           '```\n'
+           '---\n'
+           'Here is some information about how the dates work:\n'
+           '- If an ambiguous date format is used, **the English one will have preference** (`03/01/2016` will '
+           'be read as `MM/DD/YYYY`). In case of doubt, [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) '
+           'format is recommended (`YYYY/MM/DD`).\n'
+           '- If a timezone is specified in both a date and in the optional `timezone` field, **the one in the '
+           'date will have preference**.'
+           '- You can use **different languages and locales** for dates, such as `Martes 14 de enero de 2003 a '
+           'las 13:37 CET`. Some (but not all) of them are: English, Spanish, Dutch and Russian. A full list can '
+           'be found [here](https://dateparser.readthedocs.io/en/latest/#supported-languages).\n'
+           '- The default timezone is **the server\'s**. However, it can be specified at the end of each date, '
+           'in both numerical (`+01:00`) or abbreviated format (`CET`).')
+
+    reply_message(msg, message, client)
+
+def send_parsing_error_message(err, message, client):
+    error = ''
+
+    if err.error_id == 1:
+        error = ('Unknown message format.\n\n'
+                 'Usage examples:\n'
+                 '```\n'
+                 '@gcalendar Meeting with John | 2017/03/14 13:37 | 2017/03/14 15:00:01 | GMT\n'
+                 '@gcalendar Trip to LA | tomorrow\n'
+                 '```\n'
+                 'Send `@gcalendar help` for detailed usage instructions.')
+    elif err.error_id == 2:
+        error = ('Unknown date format.\n\n'
+                 'Send `@gcalendar help` for detailed usage instructions.')
+    elif err.error_id == 3:
+        error = ('Unknown timezone.\n\n'
+                 'Please, use a numerical (`+01:00`) or abbreviated (`CET`) timezone.\n'
+                 'Send `@gcalendar help` for detailed usage instructions.')
+
+    reply_message(error, message, client)
+
+class GCalendarHandler(object):
+    '''
+    This plugin facilitates creating Google Calendar events.
+
+    Usage:
+        For delimited events:
+            @gcalendar <event_title> | <start_date> | <end_date> | <timezone> (optional)
+        For full-day events:
+            @gcalendar <event_title> | <start_date>
+
+    The "event-title" supports all characters but pipes (|)
+
+    Timezones can be specified in both numerical (+00:00) or abbreviated
+    format (UTC).
+    If no timezone is specified, the server's will be considered as default.
```

Comment:
Would something like "...the local one, taken from the server..." sound better, maybe?